### ISSUE TITLE
[Feature]: implement react/jsx-no-useless-fragment

### ISSUE THREAD

**Issue by fi3ework** [web-infra-dev/rslint#1937]
### What problem does this feature solve?

We are migrating the [storybook-rsbuild](https://github.com/rstackjs/storybook-rsbuild) monorepo from Biome to Rslint. Almost all of our previous lint policy mapped 1:1 onto Rslint's existing rules — the one gap left on the React side is `react/jsx-no-useless-fragment`.

The rule is already listed (unchecked) in the eslint-plugin-react tracking issue #225, so this is a request to prioritize it. Reasons it pulls ahead of the rest of the backlog:

- It is widely deployed: eslint-config-airbnb enables it at `error` level ([rules/react.js](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/rules/react.js)), and it has been ported by Biome (as [`noUselessFragments`](https://biomejs.dev/linter/rules/no-useless-fragments/)) and by Oxlint ([react/jsx-no-useless-fragment](https://oxc.rs/docs/guide/usage/linter/rules/react/jsx-no-useless-fragment)). Migrations from either tool hit this gap immediately.
- Original rule for reference: [eslint-plugin-react docs](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-useless-fragment.md).

### What does the proposed API of configuration look like?

Same as upstream eslint-plugin-react, no new API surface:

```js
rules: {
  'react/jsx-no-useless-fragment': ['warn', { allowExpressions: true }],
}
```

Options to match upstream: `allowExpressions` (default `false`).
